core/conn/jdbc_type2/native/SrvrKds.cpp

```diff
@@ -193,3 +196,43 @@ void kdsCopyToSQLValueSeq(SQLValueList_def *SQLValueList,
 	SQLValueList->_length++;
 	FUNCTION_RETURN_VOID((NULL));
 }
+void kdsCopySQLErrorExceptionAndRowCount(
+							odbc_SQLSvc_SQLError *SQLError, 
+							char *msg_buf,
+							long sqlcode,
+							char *sqlState,
+							long currentRowCount)
+{
+	SRVRTRACE_ENTER(FILE_KDS+6);
+	ERROR_DESC_def *SqlErrorDesc;
+	size_t	len;
+
+	len = strlen(msg_buf);
+	// Allocate String Buffer
+	
+	SqlErrorDesc = (ERROR_DESC_def *)SQLError->errorList._buffer + SQLError->errorList._length;
+	SqlErrorDesc->errorText = new char[len+1];
+	if (SqlErrorDesc->errorText == NULL)
+	{
+		exit(0);
```

Comment:
T2 do has no memory allocation failure management, even in its wrapped MEMORY_ALLOC macros, it always simply exit. Memory control enhancement is another thing that we are trying to do. In the past in MXOSRVR, this kind of error is written into event repository, but now we don't have repository enabled yet, so we just keep it this way shortly.